### PR TITLE
Feature/wip dag factory generator

### PR DESCRIPTION
**Pull request by BAntonellini**
This merge introduces:

1. **AirbyteGenerator** and **AirbyteDbtGenerator** (iherits from the first one) inside 'dagfactory/generators/'
2. New **'dbt'** folder at the root of the project, which enables dbt features and work.
3. Modification in 'scripts/**entrypoint.sh**': add (via bash) the Airflow connection pointing to Airbyte
4. Some minor modifications on **dagbuilder.py** to improve work with generators

### CODE REVIEW

File: dagfactory/generators/airbyte_generator.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+from airflow.models.base import Base
+from requests.exceptions import RequestException
+from ..dagfactory import DagBuilder
+from airflow.utils.dates import days_ago
+from airflow.providers.airbyte.operators.airbyte import AirbyteTriggerSyncOperator
+from airflow.hooks.base import BaseHook
+import subprocess, requests, json
+
+# IMPORTS FOR TESTING
+import ipdb
```
Comment (ssassi) [inline dagfactory/generators/airbyte_generator.py]: Remove unused imports.

File: dagfactory/generators/airbyte_generator.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            airbyte_connection = BaseHook.get_connection(airbyte_connection_name)
+        except KeyError:
+            raise AirbyteGeneratorException("`airflow_connection_id` is missing in DAG's configuration YAML file")
+
+        AIRBYTE_API_URL = f"http://{airbyte_connection.host}:{airbyte_connection.port}/api/v1/"
+        AIRBYTE_API_ENDPOINT_LIST_ENTITY = AIRBYTE_API_URL + "{entity}/list"
+        AIRBYTE_API_ENDPOINT_LIST_WORKSPACES = AIRBYTE_API_ENDPOINT_LIST_ENTITY.format(entity="workspaces")
+        AIRBYTE_API_ENDPOINT_LIST_CONNECTIONS = AIRBYTE_API_ENDPOINT_LIST_ENTITY.format(entity="connections")
+        AIRBYTE_API_ENDPOINT_LIST_DESTINATIONS = AIRBYTE_API_ENDPOINT_LIST_ENTITY.format(entity="destinations")
+
```
Comment (ssassi) [inline dagfactory/generators/airbyte_generator.py]: I wouldn't capitalize these variables, as they are not reusable constants, just variables to be used in this scope.

File: dbt/dbt_cmd_test.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from typing import Dict
```
Comment (ssassi) [inline dbt/dbt_cmd_test.py]: What's the purpose of this file?

File: dbt/dbt_modules/dbt_date/.gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (ssassi) [inline dbt/dbt_modules/dbt_date/.gitignore]: `dbt/dbt_modules` should be ignored completely

**Review by ssassi — CHANGES_REQUESTED** [review]
Please apply python black in your python files. Ignore the usual folders to be ignored of the dbt project.

File: dagfactory/generators/__init__.py [hunk 5]
```diff
@@ -1,2 +1,7 @@
-"""Modules and methods to export for easier access"""
-from .dagfactory import DagFactory
+"""Generators."""
+from .airbyte_generator import AirbyteGenerator, AirbyteDbtGenerator
+
+
+_AIRBYTE_HOST = 'localhost'
```
Comment (ssassi) [inline dagfactory/generators/__init__.py]: Remove these hardcoded values

File: dagfactory/dagbuilder.py [hunk 6]
```diff
@@ -4,6 +4,7 @@
 
 import os
 import importlib
+import ipdb
```
Comment (ssassi) [inline dagfactory/dagbuilder.py]: Remove unused imports

File: dagfactory/dagfactory.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
 """Module contains code for loading a DagFactory config and generating DAGs"""
-import os
+import os, ipdb
```
Comment (ssassi) [inline dagfactory/dagfactory.py]: Remove unused import

**Review by ssassi — CHANGES_REQUESTED** [review]
Some additional findings

File: Dockerfile [hunk 8]
```diff
@@ -33,11 +33,15 @@ RUN set -ex \
...
 
-RUN pip install apache-airflow==${AIRFLOW_VERSION} ipdb
+RUN pip install apache-airflow==${AIRFLOW_VERSION} ipdb dbt wtforms==2.3.3
 RUN mkdir -p /root/airflow/dags
 
 ADD . /app
 ADD examples/example_basic.py /root/airflow/dags/example_basic.py
+ADD examples/example_airflow_dbt.py /root/airflow/dags/example_airflow_dbt.py
+ADD examples/example_airbyte_operator.py /root/airflow/dags/example_airbyte_operator.py
+ADD profiles.yml /home/root/.dbt/profiles.yml
```
Comment (ssassi) [inline Dockerfile]: `/home/root/`, Is this being used?

**Comment by BAntonellini** [comment]
PR found issues fixed: dbt ignores, unused imports, black linting and variable case correction

File: README.md [hunk 9]
```diff
@@ -85,7 +85,7 @@ Contributions are welcome! Just submit a Pull Request or Github Issue.
 
 ```bash
 docker build -t dag-factory .
-docker run -d -p "8080:8080" -v "$(pwd):/app" dag-factory
+docker run -d -p "8080:8080" -v "$(pwd):/app" --network main_network dag-factory
```
Comment (ssassi) [inline README.md:90]: I would add a comment here, explaining that Airbyte needs to be ran in the same docker network

**Review by ssassi — CHANGES_REQUESTED** [review]
Some additional notes

File: .gitignore [hunk 10]
```diff
@@ -109,3 +109,10 @@ pip-selfcheck.json
 
 # IntelliJ
 .idea/*
+
+# dbt
+dbt/
+target/
+dbt_modules/
+.config/.user.yml
+/profiles.yml
```
Comment (ssassi) [inline .gitignore]: if you have `dbt/` ignored, we don't need the rest.

File: Dockerfile (hunk 8 above)
Comment (BAntonellini) [reply Dockerfile]: Both usages of `/home/root/` were removed as part of `dbt/` removal

**Review by ssassi — APPROVED** [review]

**Pull request merged**